tests/passw_hardening/test_passw_hardening.py

```diff
@@ -0,0 +1,682 @@
+PAM_PASSWORD_CONF_SPECIAL_LETTER_ONLY_EXPECTED = CURR_DIR+'/sample/passw_hardening_special_letter/common-password'
+PAM_PASSWORD_CONF_LEN_MIN_ONLY_EXPECTED = CURR_DIR+'/sample/passw_hardening_min_len/common-password'
+PAM_PASSWORD_CONF_OUTPUT = CURR_DIR+'/output/login.def'
+
+# Linux login.def default values (password hardening disable)
+LINUX_DEFAULT_PASS_MAX_DAYS = 99999
+LINUX_DEFAULT_PASS_WARN_AGE = 7
+
+
+AGE_DICT = { 'MAX_DAYS': {'REGEX_DAYS': 'grep \'^PASS_MAX_DAYS[ \\t]*\'', 'DAYS': 'max_days', 'CHAGE_FLAG': '-M '},
+            'WARN_DAYS': {'REGEX_DAYS': 'grep \'^PASS_WARN_AGE[ \\t]*\'', 'DAYS': 'warn_days', 'CHAGE_FLAG': '-W '}
+            }
+
+# users
+USERNAME_STRONG = 'user_strong_test'
+USERNAME_SIMPLE_0 = 'user_simple_0_test'
+USERNAME_SIMPLE_1 = 'user_simple_1_test'
+USERNAME_ONE_POLICY = 'user_one_policy_test'
+USERNAME_AGE = 'user_test'
+USERNAME_HISTORY = 'user_history_test'
+USERNAME_LEN_MIN = 'user_test'
+
+
+class PasswHardening:
+    def __init__(self, state='disabled', expiration='100', expiration_warning='15', history='12',
+                        len_min='8', reject_user_passw_match='true', lower_class='true',
+                        upper_class='true', digit_class='true', special_class='true'):
+
+        self.policies = {
+                       "state": state,
+                       "expiration": expiration,
+                       "expiration-warning": expiration_warning,
+                       "history-cnt": history,
+                       "len-min": len_min,
+                       "reject-user-passw-match": reject_user_passw_match,
+                       "lower-class": lower_class,
+                       "upper-class": upper_class,
+                       "digits-class": digit_class,
+                       "special-class": special_class
+        }
+
+
+def run_diff(file1, file2):
+    try:
+        diff_out = subprocess.check_output('diff -ur {} {} || true'.format(file1, file2), shell=True)
+        return diff_out
+    except subprocess.CalledProcessError as err:
+        syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+        return -1
+
+def run_remote_command(duthost, command):
+    '''Running cmd from remote (switch).
+    If the command fail, the test module will not crash, the function will return false, other true'''
+    cmd = None
+    try:
+        logging.debug('DUT command = {}'.format(command))
+        cmd = duthost.shell(command)
+        logging.debug('command output lines = {}'.format(cmd["stdout_lines"]))
+        return True, cmd["stdout_lines"]
+    except Exception as e:
+        logging.error('command={} fail, error={}'.format(command, e))
+        return False, e.results['stderr_lines']
+
+def config_user(duthost, username, password=None, mode='add'):
+    username = username.strip()
+    cmd_res = False, False
+    if mode == 'add':
```

Comment:
i suggest the next:
command = "user{} {}".format(mode, username)
cmd_res = run_remote_command(duthost, command)
if add:
     change_user_pass